go/mysql/server.go

```diff
@@ -114,6 +120,11 @@ type Listener struct {
 
 	// Incrementing ID for connection id.
 	connectionID uint32
+
+	shutdownOnce      sync.Once
```

Comment:
the `sync.Once` may be overkill here

Reply:
As mentioned in the `drainListener` code, none of this is required. If you just closed the listener, the underlying network code already does this type of thing.

Your existing connections will still be open, which you 'll have to drain.